docs/assets/buffer2.js

```diff
@@ -0,0 +1,332 @@
+            if (buffered) results.push(buffered)
+          })
+        }
+      })
+      return featureCollection(results)
+  }
+  return bufferFeature(
+    geojson,
+    radius,
+    units,
+    steps,
+    endCapStyle,
+    joinStyle,
+    mitreLimit,
+    singleSided
+  )
+}
+
+/**
+ * Buffer single Feature/Geometry
+ *
+ * @private
+ * @param {Feature<any>} geojson input to be buffered
+ * @param {number} radius distance to draw the buffer
+ * @param {string} [units='kilometers'] any of the options supported by turf units
+ * @param {number} [steps=8] number of steps
+ * @param {number} [endCapStyle=1] end cap style (1 = round, 2 = flat, 3 = square)
+ * @param {number} [joinStyle=1] join style (1 = round, 2 = mitre, 3 = bevel)
+ * @param {number} [mitreLimit=5] mitre limit ratio
+ * @param {boolean} [singleSided=false] whether to buffer just one side of the input line
+ * @returns {Feature<Polygon|MultiPolygon>} buffered feature
+ */
+function bufferFeature (geojson, radius, units, steps, endCapStyle, joinStyle, mitreLimit, singleSided) {
+  const properties = geojson.properties || {}
+  const geometry = geojson.type === 'Feature' ? geojson.geometry : geojson
+
+  // Geometry Types faster than jsts
+  if (geometry.type === 'GeometryCollection') {
+    const results = []
+    geomEach(geojson, function (geometry) {
+      const buffered = bufferFeature(
+        geometry,
+        radius,
+        units,
+        steps,
+        endCapStyle,
+        joinStyle,
+        mitreLimit,
+        singleSided
+      )
+      if (buffered) results.push(buffered)
+    })
+    return featureCollection(results)
+  }
+
+  // Project GeoJSON to Azimuthal Equidistant projection (convert to Meters)
+  const projection = defineProjection(geometry)
+
+  const projected = {
+    type: geometry.type,
+    coordinates: projectCoords(geometry.coordinates, projection)
+  }
+
+  // GEOS buffer operation
+  const isBufferWithParams = endCapStyle || joinStyle || mitreLimit || singleSided
+  let bufferParamsPtr
+  if (isBufferWithParams) {
+    bufferParamsPtr = GEOSFunctions.GEOSBufferParams_create()
+    if (endCapStyle) {
+      GEOSFunctions.GEOSBufferParams_setEndCapStyle(bufferParamsPtr, endCapStyle)
+    }
+    if (joinStyle) {
+      GEOSFunctions.GEOSBufferParams_setJoinStyle(bufferParamsPtr, joinStyle)
+    }
+    if (mitreLimit) {
+      GEOSFunctions.GEOSBufferParams_setMitreLimit(bufferParamsPtr, mitreLimit)
+    }
+    if (steps) {
+      GEOSFunctions.GEOSBufferParams_setQuadrantSegments(bufferParamsPtr, steps)
+    }
+    if (singleSided) {
+      GEOSFunctions.GEOSBufferParams_setSingleSided(bufferParamsPtr, singleSided)
+    }
+  }
+  // create a GEOS object from the GeoJSON
+  // geojsonToPointers always returns an array of pointers
+  // const geomPtr = GEOSGeomFromWKT(stringify(projected));
+  // const wkb = Geometry.parseGeoJSON(projected).toWkb()
+  // const geomPtr = GEOSGeomFromWKB(wkb)
+  const geomPtr = jsonGeomToGeosGeom(projected)
```

Comment:
This is nitpicking, but I'd prefer the function to be called geojsonToGeosGeom, as to me jsonGeom sounds like it's some custom json geometry format

Reply:
Okay, sure.
In `src/helpers/geojson.mjs`, there are 2 private (non exported) methods at the bottom, so it may be better to rename as follows.
- `jsonCoordsToGeosCoordSeq` => `geojsonCoordsToGeosCoordSeq`
- `geosCoordSeqToJsonCoords` => `geosCoordSeqToGeojsonCoords`